### PR TITLE
Feature/current member management

### PR DESCRIPTION
**Pull request by islomN**

### CODE REVIEW

File: TipCatDotNet.Api/Infrastructure/Validations.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+namespace TipCatDotNet.Api.Infrastructure
+{
+    public static class Validations
+    {
+        public static bool IsEmail(string param, bool nullable = true)
```
Comment (kirillta) [inline TipCatDotNet.Api/Infrastructure/Validations.cs:8]: Did we need it? We always may confirm email by sending something to it

File: TipCatDotNet.Api/Models/HospitalityFacilities/MemberInfoResponse.cs [hunk 2]
```diff
@@ -6,13 +6,24 @@ namespace TipCatDotNet.Api.Models.HospitalityFacilities
 {
     public readonly struct MemberInfoResponse
     {
+        public MemberInfoResponse(int id, string firstName, string lastName, string? email, MemberPermissions permissions, string? avatarUrl)
```
Comment (kirillta) [inline TipCatDotNet.Api/Models/HospitalityFacilities/MemberInfoResponse.cs:9]: Why do we need two different constructors?

File: TipCatDotNet.Api/Services/HospitalityFacilities/IMemberService.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ public interface IMemberService
     {
         Task<Result<MemberInfoResponse>> AddCurrent(string? id, MemberPermissions permissions, CancellationToken cancellationToken = default);
         Task<Result<MemberInfoResponse>> GetCurrent(MemberContext memberContext, CancellationToken cancellationToken = default);
-        Task<Result<MemberInfoResponse>> UpdateCurrent(string? id,MemberUpdateRequest request, CancellationToken cancellationToken = default);
-        Task<Result<MemberAvatarResponse>> UpdateAvatar(string? id, MemberAvatarRequest request, CancellationToken cancellationToken = default);
+        Task<Result<MemberInfoResponse>> UpdateCurrent(MemberContext? memberContext, MemberUpdateRequest request, CancellationToken cancellationToken = default);
```
Comment (kirillta) [inline TipCatDotNet.Api/Services/HospitalityFacilities/IMemberService.cs:13]: Just `Update`. There is no difference, who you update

File: TipCatDotNet.Api/Infrastructure/DirectoryExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿
+using System.IO;
+
+namespace TipCatDotNet.Api.Infrastructure
+{
+    public static class DirectoryHelper
+    {
+        public static void CreateDirectory(string path)
```
Comment (kirillta) [inline TipCatDotNet.Api/Infrastructure/DirectoryExtensions.cs:8]: We will not store files on a disk. Will use S3 storage instead

**Review by kirillta — CHANGES_REQUESTED** [review]

File: TipCatDotNet.Api/Infrastructure/Validations.cs (hunk 1 above)
Comment (islomN) [reply TipCatDotNet.Api/Infrastructure/Validations.cs:8]: This was only added to check the data entry is correct

**Pull request closed without merging**